src/snowflake/cli/plugins/rpc/manager.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import annotations
+
+import importlib
+import inspect
+import os
+import pkgutil
+from typing import Optional, TypedDict
+
+from pygls.server import LanguageServer
+from snowflake import connector
+from snowflake.cli.api.sql_execution import SqlExecutionMixin
+from snowflake.snowpark import Session
+
+MAX_WORKERS = 5
+
+
+class ConnectionParams(TypedDict):
+    sessionToken: Optional[str]
+    masterToken: Optional[str]
+    account: Optional[str]
+    host: Optional[str]
+    port: Optional[str]
+    protocol: Optional[str]
+
+
+class LSPPluginContext:
+    """
+    Contains context for LSP Plugins
+    """
+
+    def __init__(self) -> None:
+        self._connection = self.create_connection()
+
+    def get_connection(self):
+        return self._connection
+
+    def update_connection(self, params: ConnectionParams):
+        self._connection = self._create_connection(params)
+
+    def _create_connection(self, params: ConnectionParams):
+        session_token = params["sessionToken"]
+        master_token = params["masterToken"]
+        account = params["account"]
+        host = params["host"]
+        port = params["port"]
+        protocol = params["protocol"]
+        if session_token and master_token:
+            try:
+                conn = connector.connect(
+                    account=account,
+                    host=host,
+                    port=port,
+                    protocol=protocol,
+                    session_token=session_token,
+                    master_token=master_token,
+                    server_session_keep_alive=True,
+                )
+            except connector.ProgrammingError as e:
+                error_message = str(e)
+                # "251005: User is empty" indicates the session token parameter doesn't exist on the current snowflake-python-connector version
+                if "251005" in error_message:
+                    raise connector.ProgrammingError(
+                        "Failed to use the active connection from the Snowflake extension. Requires snowflake-python-connector version 3.7.0 or higher."
+                    )
+                else:
+                    raise
+            return conn
+        else:
+            conn = connector.connect()
+            return conn
```

Comment:
Why not reuse cli logic fro building a connection?

Reply:
Thanks, I will do that.